### PR TITLE
Update fourmolu, ghcid and hls. Update haskellNix and iohkNix flakes

### PR DESCRIPTION
**Pull request by lehins**
# Description

Update tools

# Checklist

- [x] Commit sequence broadly makes sense and commits have useful messages
- [ ] New tests are added if needed and existing tests are updated
- [ ] When applicable, versions are updated in `.cabal` and `CHANGELOG.md` files according to the
      [versioning process](https://github.com/input-output-hk/cardano-ledger/blob/master/RELEASING.md#versioning-process).
- [ ] The version bounds in `.cabal` files for all affected packages are updated. **If you change the bounds in a cabal file, that package itself must have a version increase.** (See [RELEASING.md](https://github.com/input-output-hk/cardano-ledger/blob/master/RELEASING.md#versioning-process))
- [ ] All visible changes are prepended to the latest section of a `CHANGELOG.md` for the affected packages. **New section is never added with the code changes.** (See [RELEASING.md](https://github.com/input-output-hk/cardano-ledger/blob/master/RELEASING.md#changelogmd))
- [x] Code is formatted with [`fourmolu`](https://github.com/fourmolu/fourmolu) (use `scripts/fourmolize.sh`)
- [x] Cabal files are formatted (use `scripts/cabal-format.sh`)
- [x] [`hie.yaml`](https://github.com/input-output-hk/cardano-ledger/blob/master/hie.yaml) has been updated (use `scripts/gen-hie.sh`)
- [x] Self-reviewed the diff

### CODE REVIEW

File: .github/workflows/haskell.yml [hunk 1]
```diff
@@ -142,8 +142,9 @@ jobs:
 
     - name: Install fourmolu
       run: |
+        FOURMOLU_VERSION="0.14.0.0"
```
Comment (aniketd) [inline .github/workflows/haskell.yml:145]: I'm surprised this didn't require an `export` :slightly_smiling_face:

**Review by aniketd — APPROVED** [review]
Grand! :raised_hands:

File: .github/workflows/haskell.yml (hunk 1 above)
Comment (lehins) [reply .github/workflows/haskell.yml:145]: Exporting is needed only if you want the environment variable be available in the subprocess. Here, however, we just do interpolation directly in the same bash session. See the difference:

```shell
> FOO="foo"
> echo "$FOO"
foo
> bash -c 'echo "$FOO"'

> export BAR="bar"
> echo "$BAR"
bar
> bash -c 'echo "$BAR"'
bar
```

Comment (aniketd) [reply .github/workflows/haskell.yml:145]: Ah! Got it! Thank you @lehins :bow:

**Pull request merged**